### DISCUSSION TITLE
Text Layout / Reading Order detection

### DISCUSSION THREAD

**Discussion by nrodnova** [explosion/spaCy#13016] [Help: Model Advice] [answered]
Hello,

I was wondering if anyone tried to detect *text* document layout. Think a newspaper article or ACL paper copied/pasted in an ASCII file. Example:
```
This is a sentence         This is a sentence 
# 1.                       # 2.
```
When simply read as a text, line by line, the document will be interpreted as
```
This is a sentence This is a sentence #1. #2.
```
Obviously, the desired text is:
```
This is a sentence # 1. This is a sentence #2.
```
There are a lot of papers describing this task as part of OCR, i.e. detecting text layout on images. I am looking for resources on how to approach this on the text data. Any insights will be really appreciated.

**Comment by shadeMe** [comment, marked as answer]
Hi, that's a fairly general question about NLP methods - we try to keep this support forum focused on specific questions about spaCy and its related libraries.

**Reply by nrodnova** [reply]
Well, I am trying to get it accomplished using spaCy :)

**Reply by shadeMe** [reply]
spaCy doesn't have a concept of text layout/sequence order as such - it works with `Doc` objects which encapsulate a single sequence of strings. While the input itself can contain logical segments such as sections, etc, spaCy only provides a tool to detect sentence boundaries. You can use the library to split the text into sentences, but you'll need to employ a bespoke ML model (or apply algorithmic methods) to predict their order.

**Reply by nrodnova** [reply]
Hi @shadeMe, thank you for your response. I am pretty familiar with spaCy, and I understand this task doesn't fit in the paradigm. I was just wondering if any of spaCy power users managed to solve the problem. I will do more research and report back if I end up implementing something worthy.
